cmd/internal/register/maps.go

```diff
@@ -103,27 +106,52 @@ func Maps(a *atlas.Atlas, maps []config.Map, providers map[string]provider.Tiler
 				return ErrProviderNotFound{providerLayer[0]}
 			}
 
+			fmt.Println("printing provider:", provider)
+
 			// read the provider's layer names
 			layerInfos, err := provider.Layers()
+
 			if err != nil {
 				return ErrFetchingLayerInfo{
 					Provider: providerLayer[0],
 				}
 			}
 
 			// confirm our providerLayer name is registered
-			var found bool
+			// this must be an array because auto providers may have multiple layers that match regex
+			var provLayers []string
 			var layerGeomType tegola.Geometry
-			for i := range layerInfos {
-				if layerInfos[i].Name() == providerLayer[1] {
-					found = true
-
-					// read the layerGeomType
-					layerGeomType = layerInfos[i].GeomType()
-					break
+			var isregex bool
+		LayerLoop:
+			for _, info := range layerInfos {
+				if providerLayer[1] == "*" {
+					// return all the layers:
+					provLayers = append(provLayers, info.Name())
+					layerGeomType = info.GeomType()
+					continue
+				}
+				// check to see if string contains regex
+				isregex = len(strings.Split(regexp.QuoteMeta(providerLayer[1]), "\\")) > 1
+				if isregex {
+					r, err := regexp.Compile("^" + providerLayer[1])
+					if err != nil {
+						log.Printf("Error when parsing regex (layer: %v): %v", info.Name(), err)
+						continue LayerLoop // add a Providers label at 111
+					}
+					if !r.MatchString(info.Name()) {
+						continue
+					}
+					provLayers = append(provLayers, info.Name())
+					layerGeomType = info.GeomType()
+				} else {
+					if info.Name() == providerLayer[1] {
+						provLayers = append(provLayers, info.Name())
+						layerGeomType = info.GeomType()
+					}
```

Comment:
I'm not a fan of else if it's not needed. I would write this as:
```go
switch {
   case isregex:
	r, err := regexp.Compile("^" + providerLayer[1])
	if err != nil {
		log.Printf("Error when parsing regex (layer: %v): %v", info.Name(), err)
		continue LayerLoop // add a Providers label at 111
	}
	if !r.MatchString(info.Name()) {
		continue
	}
    case info.Name() == providerLayer[1]:
        // do nothing
     default:
           // ignore this layer
          continue

}

provLayers = append(provLayers, info.Name())
layerGeomType = info.GeomType()

```

But this is just a style preference